src/Behat/Behat/Util/StrictRegex.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Behat\Behat\Util;
+
+use InvalidArgumentException;
+
+/**
+ * @internal
+ */
+final class StrictRegex
+{
+    /**
+     * @param string|list<string> $pattern
+     * @param string|list<string> $replacement
+     *
+     * @throws RegexException if the regex fails
+     * @throws InvalidArgumentException if passing pattern & replacements as arrays and the number of entries does not match
+     */
+    public static function replace(
+        string|array $pattern,
+        string|array $replacement,
+        string $subject,
+        int $limit = -1,
+    ): string {
+        if (is_array($pattern) && is_array($replacement) && count($pattern) !== count($replacement)) {
+            // If the number of replacements does not match the number of patterns, PHP treats missing replacements as
+            // empty strings. However, it is safer to force the caller to be explicit about this.
+            throw new InvalidArgumentException(
+                sprintf('Expected %d entries in $replacement array, got %d', count($pattern), count($replacement)),
+            );
+        }
+
+        $result = self::callWithErrorCapture(fn (): ?string => preg_replace($pattern, $replacement, $subject, $limit));
+        if ($result === null) {
+            throw new RegexException('Regex failed: '.preg_last_error_msg(), preg_last_error());
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param string|list<string> $pattern
+     * @param callable(array<mixed,string>):string $callback
+     *
+     * @throws RegexException if the regex fails
+     */
+    public static function replaceCallback(array|string $pattern,
+        callable $callback,
+        string $subject,
```

Comment:
why this weird splitting of the method signature which is not fully multiline ? This does not match the PSR-12 code styles, which expects being either single-line or having one argument per line, doing it fully.

Reply:
Accidental - started as a copy & paste from the native php stub, wasn't auto-formatted correctly & I didn't spot it. Thanks for flagging.